relayer/relays/execution/main.go

```diff
@@ -0,0 +1,70 @@
+package execution
+
+import (
+	"context"
+
+	"github.com/snowfork/snowbridge/relayer/chain/ethereum"
+	"github.com/snowfork/snowbridge/relayer/chain/parachain"
+	"github.com/snowfork/snowbridge/relayer/crypto/sr25519"
+	"github.com/snowfork/snowbridge/relayer/relays/execution/config"
+	"github.com/snowfork/snowbridge/relayer/relays/execution/message"
+	"golang.org/x/sync/errgroup"
+)
+
+type Relay struct {
+	config  *config.Config
+	keypair *sr25519.Keypair
+}
+
+func NewRelay(
+	config *config.Config,
+	keypair *sr25519.Keypair,
+) *Relay {
+	return &Relay{
+		config:  config,
+		keypair: keypair,
+	}
+}
+
+func (r *Relay) Start(ctx context.Context, eg *errgroup.Group) error {
+	paraconn := parachain.NewConnection(r.config.Sink.Parachain.Endpoint, r.keypair.AsKeyringPair())
+	ethconn := ethereum.NewConnection(r.config.Source.Ethereum.Endpoint, nil)
+
+	err := paraconn.Connect(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = ethconn.Connect(ctx)
+	if err != nil {
+		return err
+	}
+
+	writer := parachain.NewParachainWriter(
+		paraconn,
+		r.config.Sink.Parachain.MaxWatchedExtrinsics,
+	)
+
+	err = writer.Start(ctx, eg)
+	if err != nil {
+		return err
+	}
+
+	messages, err := message.New(
+		ctx,
+		eg,
+		writer,
+		&r.config.Source,
+		ethconn,
```

Comment:
This is a great point! It's actually a better practice to inject the dependency instances rather than creating it in the package, because it makes testing easier (can inject mock interfaces instead of the real connection to Ethereum). I've pulled out the dependencies and inject them now. https://github.com/Snowfork/snowbridge/pull/721/commits/4a2a0d6015a4b431aa494924cead6ce3aaf7fc1e

Reply:
oh right, makes a lot of sense for testing 👌